tests/ValueParsers/TimeParserTest.php

```diff
@@ -66,372 +65,405 @@ public function validInputProvider() {
+					'-0000000000000001-01-05T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
 				$noPrecOpts,
 			),
 
 			'+1999-00-00T00:00:00Z' => array(
-				TimeValue::newFromArray( array(
-					'time' => '+0000000000001999-00-00T00:00:00Z',
-					'timezone' => 0,
-					'before' => 0,
-					'after' => 0,
-					'precision' => TimeValue::PRECISION_YEAR,
-					'calendarmodel' => TimeFormatter::CALENDAR_GREGORIAN
-			) ),
+				new TimeValue(
+					'+0000000000001999-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
 				$noPrecOpts,
 			),
 			'+2000-00-00T00:00:00Z' => array(
-				TimeValue::newFromArray( array(
-					'time' => '+0000000000002000-00-00T00:00:00Z',
-					'timezone' => 0,
-					'before' => 0,
-					'after' => 0,
-					'precision' => TimeValue::PRECISION_YEAR,
-					'calendarmodel' => TimeFormatter::CALENDAR_GREGORIAN
-			) ),
+				new TimeValue(
+					'+0000000000002000-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
 				$noPrecOpts,
 			),
 			'+2010-00-00T00:00:00Z' => array(
-				TimeValue::newFromArray( array(
-					'time' => '+0000000000002010-00-00T00:00:00Z',
-					'timezone' => 0,
-					'before' => 0,
-					'after' => 0,
-					'precision' => TimeValue::PRECISION_YEAR,
-					'calendarmodel' => TimeFormatter::CALENDAR_GREGORIAN
-			) ),
+				new TimeValue(
+					'+0000000000002010-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional sign character
+			'2015-01-01T00:00:00Z' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional time zone
+			'2015-01-01T00:00:00' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Actual minus character from Unicode; roundtrip with TimeDetailsFormatter
+			"\xE2\x88\x922015-01-01T00:00:00" => array(
+				new TimeValue(
+					'-0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional second
+			'2015-01-01T00:00' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional hour and minute
+			'2015-01-01' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Years < 59 require at least hour and minute
```

Comment:
Why? A single number is *walways* a year. Two numbers are *always* year and month (though it amybe hard to tell which is which). We need to be able to conveniently enter 44-03-15 BC to indicate the time of Ceasar's death, or simply 14 to indicate the Year Augustus died. Requiring a time of day there would not be good...